core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -129,6 +129,13 @@ trait Endpoint[A] { self =>
     override def toString = self.toString
   }
 
+  def flatMap[B](fn: A => Endpoint[B]): Endpoint[B] = new Endpoint[B] {
+    override def apply(input: Input): Option[(Input, () => Future[Output[B]])] = for {
+      (r1, ao) <- self(input)
+      (r2, bo) <- fn(Await.result(ao()).value)(r1)
```

Comment:
> as long as there is no Await.result in the code

Yeah.. I tried but it was too adventurous for me so I compromise with await, that blocks of course.

> changing the endpoint output type to something that'd allow you to write flatMap with awaits

using awaits but not block? such a magic exists?